core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractCacheSinkProvider.java

```diff
@@ -0,0 +1,53 @@
+package org.mule.runtime.core.internal.processor.strategy;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.transaction.Transaction;
+import org.mule.runtime.core.api.transaction.TransactionCoordination;
+import reactor.core.publisher.FluxSink;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.lang.Thread.currentThread;
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+public abstract class AbstractCacheSinkProvider {
```

Comment:
this should implement an interface.

Reply:
I would name, this classes and the possible interface ReactorSinkProvider so that we have two levels, the sink from mule api and the reactor sink where the fluxSinK is actually createed.